eva_submission/submission_qc_checks.py

```diff
@@ -0,0 +1,216 @@
+                logger.error(f'{file} not found in ftp')
+            if f'{file}.tbi' not in files_in_ftp:
+                logger.error(f'{file}.tbi not found in ftp')
+            if f'{file}.csi' not in files_in_ftp:
+                logger.error(f'{file}.csi not found in ftp')
+
+            # accessioned files will not be present for human taxonomy
+            if taxonomy != 9606:
+                accessioned_file = file.replace('.vcf.gz', '.accessioned.vcf.gz')
+                if accessioned_file not in files_in_ftp:
+                    logger.error(f'{accessioned_file} not found in ftp')
+                if f'{accessioned_file}.tbi' not in files_in_ftp:
+                    logger.error(f'{accessioned_file}.tbi not found in ftp')
+                if f'{accessioned_file}.csi' not in files_in_ftp:
+                    logger.error(f'{accessioned_file}.csi not found in ftp')
+
+    @retry(tries=3, delay=2, backoff=1.5, jitter=(1, 3))
+    def get_files_from_ftp(self, project_accession):
+        ftp = FTP('ftp.ebi.ac.uk', timeout=600)
+        ftp.login()
+        ftp.cwd(f'pub/databases/eva/{project_accession}')
+        logger.info(f"Trying to fetch files for study {project_accession} "
+                    f"from ftp location: ftp.ebi.ac.uk/pub/databases/eva/{project_accession}")
+        return ftp.nlst()
+
+    def check_if_job_completed_successfully(self, file_path):
+        with open(file_path, 'r') as f:
+            for line in f:
+                if 'Job: [FlowJob: [name=genotyped-vcf-job]] completed' in line or \
+                        'Job: [FlowJob: [name=aggregated-vcf-job]] completed' in line or \
+                        'Job: [SimpleJob: [name=CREATE_SUBSNP_ACCESSION_JOB]] completed' in line:
+                    job_status = line.split(" ")[-1].replace("[", "").replace("]", "").strip()
+                    if job_status == 'COMPLETED':
+                        return True
+                    elif job_status == 'FAILED':
+                        return False
+                    else:
+                        logger.error(f'Could not determine status of variant load job in file {file_path}')
+
+    def check_if_variants_were_skipped(self, file_path):
+        with open(file_path, 'r') as f:
+            for line in f:
+                if 'lines in the original VCF were skipped' in line:
+                    return line
+
+            return None
+
+    def check_for_errors_in_case_of_job_failure(self, file_name):
+        with open(file_name, 'r') as f:
+            for line in f:
+                if 'Encountered an error executing step' in line:
+                    logger.error(f'Following error was found in log file related to execution of step: \n{line}')
+
+    def check_if_accessioning_completed_successfully(self, project_accession, vcf_files, path_to_data_dir):
+        for file in vcf_files:
+            accessioning_log_files = glob.glob(f"{path_to_data_dir}/00_logs/accessioning.*{file}*.log")
```

Comment:
For all these methods that inspect logs, is there a way we can isolate only the most recent run within a log file?  Otherwise I'm worried it will be difficult to isolate the useful log lines from multiple runs.

Reply:
have changed the logic to scan through entire file and return the result for last run of job